tools/@aws-cdk/cdk-build-tools/config/eslintrc.js

```diff
@@ -163,6 +163,18 @@ module.exports = {
       }
     ],
 
+    // Relying on truthiness checks for numbers is going to cause problems
+    '@typescript-eslint/strict-boolean-expressions': [
+      'error',
+      {
+        // Mostly the defaults, with a couple of changes
+        allowNumber: false,
+        allowNullableString: true,
+        allowNullableBoolean: true,
```

Comment:
This can cause sneaky bugs, too. Anecdote: https://github.com/aws/aws-cdk/pull/21107